### DISCUSSION TITLE
How to get the entirety of custom error calldata returned from simulateContract and not just the 4 bytes?

### DISCUSSION THREAD

**Discussion by junkim012** [wevm/viem#1594] [Question]
**Question**
I'm wondering if there is a way to retrieve the full length of the custom error return data, and not just the first 4 bytes i.e. the function signature. 

**More Context**
- When `simulateContract` encounters a custom error, it is able to auto-parse the entire result and return the error name + its arguments as shown below which is great. 

```
ContractFunctionRevertedError: The contract function "borrow" reverted.
Error: UnsafePositionChange(uint256 newTotalDebtInVault, uint256 collateral, uint256 spot)
(1000210695370299417715527785000000000000000000, 0, 964823285548756497000000000)
```
- But **when the contract that throws the custom error is different from the contract whose function is being called**, `simulateContract` only seems to be able to return the first 4 bytes of the error signature. Shown below. 

```
The contract function "flashswapLeverage" reverted with the following signature: 0xf04e9d18
Unable to decode signature "0xf04e9d18" as it was not found on the provided ABI.
Make sure you are using the correct ABI and that the error exists on it.
You can look up the decoded signature here: https://openchain.xyz/signatures?query=0xf04e9d18.
```

The reason for needing the entire call data: 
- When a function call to ContractA externally calls ContractB and if ContractB throws the custom error, viem is not able to auto-parse the custom error since the `simulateContract` takes in the ContractA abi when called.  
- So since viem only returns the 4 bytes when it's not able to parse it itself, I am **not able to recover the return arguments of the custom error** using just the error signature **when there is the ABI mismatch** from the contract being called and the contract throwing the error.  

So curious how I could get the entire data! 

I've looked at https://github.com/wevm/viem/discussions/589 but doesn't address this specific problem

**Comment by MarkTangCd** [comment]
Did you use Metamask to call the method? if so. the error(the first 4 bytes) was thrown by Metamask.

**Comment by JackieXu** [comment]
Also getting this issue. Am directly using Viem in a server-side application.
